source/lac/sparsity_tools.cc

```diff
@@ -522,29 +522,31 @@ namespace SparsityTools
 
 
 #ifdef DEAL_II_WITH_MPI
-  template <class DSP_t>
-  void distribute_sparsity_pattern(DSP_t &dsp,
-                                   const std::vector<typename DSP_t::size_type> &rows_per_cpu,
-                                   const MPI_Comm &mpi_comm,
-                                   const IndexSet &myrange)
+  void distribute_sparsity_pattern
+  (DynamicSparsityPattern                                        &dsp,
+   const std::vector<typename DynamicSparsityPattern::size_type> &rows_per_cpu,
+   const MPI_Comm                                                &mpi_comm,
+   const IndexSet                                                &myrange)
   {
     const unsigned int myid = Utilities::MPI::this_mpi_process(mpi_comm);
-    std::vector<typename DSP_t::size_type> start_index(rows_per_cpu.size()+1);
+    std::vector<typename DynamicSparsityPattern::size_type> start_index(rows_per_cpu.size()+1);
     start_index[0]=0;
-    for (typename DSP_t::size_type i=0; i<rows_per_cpu.size(); ++i)
+    for (typename DynamicSparsityPattern::size_type i=0; i<rows_per_cpu.size(); ++i)
```

Comment:
You don't need the `typename` here and it is in fact wrong in C++98. Same above and possibly in other places below.